app/src/services/auth/authService.js

```diff
@@ -0,0 +1,132 @@
+"use strict";
+
+const UserRepository = require("../../repositories/user/userRepository");
+const JwtService = require("../../utils/jwtService");
+const bcrypt = require("bcrypt");
+
+class AuthService {
+  constructor(req) {
+    this.req = req;
+    this.body = req.body;
+    this.userRepository = new UserRepository();
+    this.jwtService = new JwtService();
+  }
+
+  async signUp() {
+    const { user_id, password, nickname } = this.body;
+
+    const isUserIdExist = await this.userRepository.findByUserId(user_id);
+    const isNicknameExist = await this.userRepository.findByNickname(nickname);
+
+    if (isUserIdExist && isNicknameExist) {
+      return { status: 409, success: false, data: { field: "both" } };
```

Comment:
현재 서버에서 관리하는 로우의 상태와 충돌되기 때문에 상태코드를 409로 정의하신 부분 좋습니다.
한가지 제안사항이 field 필드를 배열형태로 확장하여 사용하는건 어떨까요
both라는 필드는 실제로 존재하지 않고 해석하기 어려울거같아요

```suggestion
      return { status: 409, success: false, data: { fields: ["userId", "nickname"] } };
```

Reply:
피드백 감사합니다!
말씀해주신 대로 field를 문자열 하나로 표현하는 것보다 배열로 만들어 명확하게 전달하는 게 더 직관적이고 해석하기 쉬울 것 같습니다.
제안해주신 방식으로 전체 코드에 반영해서 수정하겠습니다!